firebase_ai_logic_showcase/lib/flutter_firebase_ai_demo.dart

```diff
@@ -13,203 +13,138 @@
-                              'https://github.com/flutter/demos/issues',
-                            ),
-                            target: LinkTarget.blank,
-                            builder: (context, followLink) => GestureDetector(
-                              onTap: followLink,
-                              child: Text(
-                                style: Theme.of(context).textTheme.bodyMedium!
-                                    .copyWith(
-                                      fontWeight: FontWeight.bold,
-                                      height: 1.15,
-                                      decoration: TextDecoration.underline,
-                                      color: Theme.of(
-                                        context,
-                                      ).colorScheme.primary,
-                                    ),
-                                'github.com/flutter/demos/issues',
-                              ),
-                            ),
-                          ),
-                        ),
-                        TextSpan(text: '.'),
-                      ],
-                    ),
-                  ),
-                  SizedBox.square(dimension: 32),
-                  Text(
-                    style: TextStyle(
-                      color: Theme.of(context).colorScheme.onSurfaceVariant,
-                    ),
-                    textAlign: TextAlign.center,
-                    'This app was made with ❤️\nby the Flutter & Firebase AI Logic Teams',
-                  ),
-                ],
-              ),
-            ),
-          ),
-        ),
-      ),
-    );
+  @override
+  void initState() {
+    super.initState();
+    _demoPages = <Widget>[
+      const ChatDemo(),
+      const LiveAPIDemo(),
+      const MultimodalDemo(),
+      ChatDemoNano(key: _chatNanoKey),
+    ];
+  }
+
+  void _onItemTapped(int index) {
+    if (index == 3 && !_nanoPickerHasBeenShown) {
+      _chatNanoKey.currentState?.showModelPicker();
+      _nanoPickerHasBeenShown = true;
+    }
+    setState(() {
+      _selectedIndex = index;
+    });
   }
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        backgroundColor: Colors.transparent,
-        leading: Padding(
-          padding: EdgeInsets.fromLTRB(16, 8, 4, 8),
-          child: Image.asset('assets/firebase-ai-logic.png'),
-        ),
-        title: Text(
-          style: Theme.of(context).textTheme.titleLarge,
-          'Flutter AI Playground',
-        ),
-        actions: [
-          Padding(
-            padding: EdgeInsets.fromLTRB(4, 8, 16, 8),
-            child: IconButton(
-              icon: Icon(Icons.info_outline),
-              onPressed: () => showMoreInfo(context),
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        if (constraints.maxWidth < 600) {
+          // Use BottomNavigationBar for smaller screens
```

Comment:
when the app gets scaled down, it regenerates the demo and re-executes the function call to change the app color. I don't think that's an intended effect?

Reply:
fixed